src/main/java/com/teragrep/pth_06/scheduler/BatchUnit.java

```diff
@@ -59,7 +59,17 @@
  * @since 08/06/2022
  * @author Mikko Kortelainen
  */
-public final class BatchSlice implements Serializable {
+public final class BatchUnit implements Serializable, Comparable<BatchUnit> {
+
+    @Override
+    public int compareTo(final BatchUnit other) {
+        if (other == null) {
+            throw new IllegalArgumentException("BatchUnit cannot be null");
```

Comment:
See the other compareTo note for a bit more clarity

Reply:
removed, NPE is ok in this case.